server/queries/queries.js

```diff
@@ -0,0 +1,104 @@
+const { Client } = require('pg');
+const Promise = require('promise');
+
+let pool;
+
+if (process.env.DATABASE_URL) {
+    pool = new Client({
+        connectionString: process.env.DATABASE_URL
+    });
+} else {
+    pool = new Client({
+        user: process.env.USER,
+        host: process.env.HOST,
+        database: 'telltales',
+        password: process.env.PASSWORD,
+        port: process.env.PORT
+    });
+}
+
+pool.connect((err) => {
+    if (err) {
+        console.error('Connection error', err.stack);
+    } else {
+        console.log("Connected");
+    }
+});
+
+// Create the table if it doesn't already exist
+/* eslint max-len: ["error", { "code": 200 }] */
+const createTableQuery = 'CREATE TABLE IF NOT EXISTS users ( id SERIAL PRIMARY KEY, name VARCHAR(15) NOT NULL, score INT );'
+pool.query(createTableQuery, 
+    (err) => {
+    if (err) {
+        console.log(err);
+    } else {
+        console.log("Created table users");
+    }
+});
+
+// Add a user to the table
+module.exports.add = (username) => {
+    return new Promise((resolve, reject) => {
+        let query = 'INSERT INTO users (name, score) VALUES ($1, $2) RETURNING id';
+        pool.query(query, [username, 0], (err, result) => {
+            if (err) {
+                reject({
+                    code: 500,
+                    msg: err
+                })
+            } else {
+                var id = result.rows[0].id;
+                resolve(id);
+            }
+        });
+    });
+}
+
+// Get user by username
+module.exports.get = (username) => {
+    return new Promise((resolve, reject) => {
+        let query = 'SELECT * FROM users WHERE name = $1';
+        pool.query(query, [username], (err, result) => {
+            if (err) {
+                reject({
+                    code: 404,
+                    msg: err
+                })
+            } else {
+                resolve(result.rows[0]);
+            }
+        });
+    });
+}
+
+// Add a user to the table
+module.exports.updateScores = (players) => {
```

Comment:
This function is getting a bit complicated, might look nicer if it was just `updateScore` and you did the looping outside?

Reply:
If I were to move it outside, it would be in controllers/index.js and I thought we should keep the logic outside this file